common/types/block.go

```diff
@@ -68,9 +68,33 @@ func GetEffectiveGenesis() LayerID {
 	return LayerID(atomic.LoadInt32(&EffectiveGenesis))
 }
 
-// Add returns the LayerID that's layers (the param passed into this method) after l (this LayerID).
-func (l LayerID) Add(layers uint16) LayerID {
-	return LayerID(uint64(l) + uint64(layers))
+// Add layers to the layer. Panics on wraparound.
+func (l LayerID) Add(layers LayerID) LayerID {
```

Comment:
`Add` (and all other new methods here) should accept a `uint16` (which should be our standard for layer-deltas). This is similar to `Time.Add()` accepting a duration and not another time ([see here](https://golang.org/pkg/time/#Time.Add)).

Reply:
the thing is that in some places two LayerIDs were added together. i didn't want to cast uint64 to uint16 just to perform "safe" arithmetic

Reply:
@noamnelke to make it safe to use i need to store those values as uint16 then

Reply:
`bufferSize` is currently an `int`

https://github.com/spacemeshos/go-spacemesh/blob/d16d26f4c68062b06ce4f5b2c8281034e2e89f6b/hare/hare.go#L71

`HDist` is currently a `LayerID`

https://github.com/spacemeshos/go-spacemesh/blob/d16d26f4c68062b06ce4f5b2c8281034e2e89f6b/tortoise/verifying_tortoise.go#L41

I think both should become `uint16`.

Alternatively, we could create a special type for this, like Go's datetime's `Duration`, but I think that's overkill and not worth the effort.

Reply:
Actually, I think both "layer duration" and the underlying layer itself (the number inside the struct) should become `uint32`. That would give us enough range for ~40,000 years (assuming 5 minute layers) and save us the need to convert types.

Reply:
While I have high aspirations for Spacemesh, I don't believe this code will run unmodified for more than 40k years 😉

Reply:
Layers should be uint32. See #2082.